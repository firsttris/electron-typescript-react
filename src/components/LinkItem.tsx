import * as React from 'react';
const {Link} = require('react-router-dom');

interface Props {
    focus:boolean,
    path:string,
    name:string
}

interface State {
    hover:boolean
}

class LinkItem extends React.Component<Props, State> {
    constructor() {
        super();
        this.state = {
            hover: false
        }
    }

    onMouseLeave():void {
        this.setState({hover: false});
    }

    onMouseEnter():void {
        this.setState({hover: true});
    }

    getLinkStyle():any {
        const linkStyle = {
            display: 'block',
            color: 'white',
            textAlign: 'center',
            padding: '14px 16px',
            textDecoration: 'none'
        };
        if (this.state.hover || this.props.focus) {
            return {...linkStyle, ...{backgroundColor: 'white', color: '#222'}};
        }
        return linkStyle;
    }

    render() {
        return (
            <li style={{float: 'left'}} onMouseLeave={this.onMouseLeave.bind(this)}
                onMouseEnter={this.onMouseEnter.bind(this)}><Link style={this.getLinkStyle()}
                                                                  to={this.props.path}>{this.props.name}</Link></li>
        );
    }
}

export default LinkItem;