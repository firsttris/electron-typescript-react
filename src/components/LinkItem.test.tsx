import * as React from 'react';
const renderer = require('react-test-renderer');
import LinkItem from "./LinkItem";
const {Router} = require('react-router-dom');
const createHistory = require('history/createMemoryHistory').default;

test('FadeCard changes the class when hovered', () => {
    const component = renderer.create(
        <Router history={createHistory()}>
            <LinkItem focus={true} path="/" name="App"/>
        </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});