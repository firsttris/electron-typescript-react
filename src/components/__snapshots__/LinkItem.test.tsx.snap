// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeCard changes the class when hovered 1`] = `
<li
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "float": "left",
    }
  }
>
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "color": "#222",
        "display": "block",
        "padding": "14px 16px",
        "textAlign": "center",
        "textDecoration": "none",
      }
    }
  >
    App
  </a>
</li>
`;

exports[`FadeCard changes the class when hovered 2`] = `
<li
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "float": "left",
    }
  }
>
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "color": "#222",
        "display": "block",
        "padding": "14px 16px",
        "textAlign": "center",
        "textDecoration": "none",
      }
    }
  >
    App
  </a>
</li>
`;

exports[`FadeCard changes the class when hovered 3`] = `
<li
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "float": "left",
    }
  }
>
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "color": "#222",
        "display": "block",
        "padding": "14px 16px",
        "textAlign": "center",
        "textDecoration": "none",
      }
    }
  >
    App
  </a>
</li>
`;
